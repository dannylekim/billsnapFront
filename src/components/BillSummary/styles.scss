@import "../../variables";

span#more__details {
    text-decoration: underline;
    color: $primary-text-color;
    float: right;
    margin-bottom: 15px;
    margin-right: 32px;
    right: 0;
  }

  h5 span#amount__owed {
    color: $secondary-price-color;
  }

  div.bill__summary {
    text-align: left;
    margin: 2em;
  }