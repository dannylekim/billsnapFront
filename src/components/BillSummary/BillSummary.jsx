import React, { Component } from "react";

import './styles.scss';

class BillSummary extends Component {
  render() {
    const { billCount, activeBill, amountToOwe } = this.props;

    return (
      <div className='bill__summary'>
        {activeBill && activeBill.id ? (
          <>
            <h5>
              Split by : {activeBill.responsible.firstName} {activeBill.responsible.lastName}
            </h5>
            <h5>Status : {activeBill.status}</h5>
            <h5>
              Amount Owed :
              <span id='amount__owed'> {activeBill.balance} $</span>{" "}
            </h5>
          </>
        ) : (
          <>
            <h5>
              Total Amount Owed :
              <span id='amount__owed'> {amountToOwe} $</span>
            </h5>
            <h5> Total of bills : {billCount} </h5>
          </>
        )}
      </div>
    );
  }
}

export default BillSummary;
