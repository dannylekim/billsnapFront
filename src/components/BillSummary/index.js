import BillSummary from './BillSummary';
import {connect} from "react-redux";
import { getTotalAmountOwe, getCountOfBills } from "../../redux/selectors/billsSelectors";


const mapStateToProps = (state) => ({
    billCount: getCountOfBills(state),
    amountToOwe: getTotalAmountOwe(state),
    activeBill: state.bills.activeBill,
  });

export default connect(mapStateToProps)(BillSummary);