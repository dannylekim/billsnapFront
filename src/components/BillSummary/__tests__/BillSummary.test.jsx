import React from "react";
import BillSummary from "../BillSummary";

describe("BillSummary", () => {
  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("BillSummary should match snap shot with empty bill", () => {
        matches(
          <BillSummary
            activeBill={{}}
          />
        );
      });

      it("BillSummary should match snap shot with empty active bill and many bills count", () => {
        matches(
          <BillSummary
            activeBill={{}}
            amountToOwe={123.00}
            billCount={100}
          />
        );
      });

      it("BillSummary should match snap shot with empty active bill and many bills count", () => {
        const mockBill = {
            id: 1,
            responsible: {
                firstName: 'Bob',
                lastName: 'Ross',
            },
            status: 'OPENED',
            balance: 20.20,
        }
        matches(
          <BillSummary
            activeBill={mockBill}
            amountToOwe={123.00}
            billCount={100}
          />
        );
      });
    });
  });

  describe("component ", () => {
  });
});
