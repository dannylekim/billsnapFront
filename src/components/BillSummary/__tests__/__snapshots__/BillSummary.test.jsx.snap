// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillSummary render snapshots ðŸ“¸ BillSummary should match snap shot with empty active bill and many bills count 1`] = `
<div
  className="bill__summary"
>
  <h5>
    Total Amount Owed :
    <span
      id="amount__owed"
    >
       
      123
       $
    </span>
  </h5>
  <h5>
     Total of bills : 
    100
     
  </h5>
</div>
`;

exports[`BillSummary render snapshots ðŸ“¸ BillSummary should match snap shot with empty active bill and many bills count 2`] = `
<div
  className="bill__summary"
>
  <h5>
    Split by : 
    Bob
     
    Ross
  </h5>
  <h5>
    Status : 
    OPENED
  </h5>
  <h5>
    Amount Owed :
    <span
      id="amount__owed"
    >
       
      20.2
       $
    </span>
     
  </h5>
</div>
`;

exports[`BillSummary render snapshots ðŸ“¸ BillSummary should match snap shot with empty bill 1`] = `
<div
  className="bill__summary"
>
  <h5>
    Total Amount Owed :
    <span
      id="amount__owed"
    >
       
       $
    </span>
  </h5>
  <h5>
     Total of bills : 
     
  </h5>
</div>
`;
